File: 250711/되돌아오기/come-back.py
N = int(input())
moves = [tuple(input().split()) for _ in range(N)]
dir = [move[0] for move in moves]
dist = [int(move[1]) for move in moves]

directions = 'EWSN'
dxy = [(0,1),(0,-1),(1,0),(-1,0)]

x = 0
y = 0
time = 0
for i in range(N):
    direction = dir[i]

    for _ in range(dist[i]):
        time += 1
        d = directions.find(direction)
        x += dxy[d][0]
        y += dxy[d][1]

        if (x == 0 and y == 0):
            print(time)
            exit()

print(-1)

File: 250711/되돌아오기/come-back.java
import java.util.Scanner;
public class Main {
    static int n;
    // 동서남북
    static int[] dx = {0,0,1,-1}, dy = {1,-1,0,0};
    static String direction = "EWSN";
    static int x = 0, y = 0;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();

        // n번 이동
        int time = 0;
        A : for(int i = 0; i < n; i++){
            char dir = sc.next().charAt(0);
            int dist = sc.nextInt();

            int d = direction.indexOf(dir);

            // 이동거리만큼 시간이 흐름
            for (int j = 1; j <= dist; j++) {
                time++;
                x = x + dx[d];
                y = y + dy[d];

                if (isStart()) break A;
            }
        }

        if (!isStart()) {
            System.out.print(-1);
        } else {
            System.out.print(time);
        }
    }

    public static boolean isStart() {
        return x == 0 && y == 0;
    }
}